ocs_ci/resiliency/resiliency_workload_config.py

```diff
@@ -0,0 +1,355 @@
+        """
+        try:
+            # Check if resiliency_config exists - must be provided
+            resiliency_config = self.config.ENV_DATA.get("resiliency_config", {})
+            if not resiliency_config:
+                raise ValueError(
+                    "Missing required 'resiliency_config' section in configuration. "
+                    "Resiliency tests require explicit configuration due to their resource-intensive nature. "
+                    "Please provide configuration via --ocsci-conf parameter. "
+                    "Example: run-ci --ocsci-conf conf/ocsci/resiliency_tests_config.yaml "
+                    "See conf/ocsci/resiliency_tests_config.yaml for configuration examples."
+                )
+
+            # Validate workload names if any are specified
+            workloads = self.get_workloads()
+            if workloads:
+                # Define valid workload types
+                valid_workloads = {
+                    self.VDBENCH,
+                    self.CNV_WORKLOAD,
+                    self.FIO,
+                    self.RGW_WORKLOAD,
+                }
+
+                # Check for invalid workload names (catches typos)
+                invalid_workloads = set(workloads) - valid_workloads
+                if invalid_workloads:
+                    raise ValueError(
+                        f"Invalid workload type(s): {', '.join(invalid_workloads)}. "
+                        f"Valid workload types are: {', '.join(sorted(valid_workloads))}"
+                    )
+            else:
+                # Empty workload list is valid for failure-injection-only scenarios
+                log.info(
+                    "No workloads configured. Running in failure-injection-only mode. "
+                    "No workload will be executed during the test."
+                )
+
+            # Validate VDBENCH config if enabled
+            if "VDBENCH" in workloads:
+                vdbench_config = self.get_vdbench_config()
+                if not vdbench_config:
+                    log.warning("VDBENCH enabled but no vdbench_config found")
+                    return True
+
+                # Check block and filesystem configs
+                block_config = vdbench_config.get("block", {})
+                filesystem_config = vdbench_config.get("filesystem", {})
+
+                if not block_config and not filesystem_config:
+                    log.warning("No block or filesystem workload configuration found")
+
+                # Validate patterns
+                for workload_type in ["block", "filesystem"]:
+                    config_section = vdbench_config.get(workload_type, {})
+                    if config_section:
+                        patterns = config_section.get("patterns", [])
+                        if not patterns:
+                            log.warning(
+                                f"No patterns configured for {workload_type} workloads"
+                            )
```

Comment:
All the check is for the log warning? in the end you are returning true if "VDBENCH" in workloads:
Will it work if none of the workloads is defined?